modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/util/CalendarUtil.java

```diff
@@ -202,11 +205,14 @@ public static JSONObject toCalendarBookingJSONObject(
 		List<CalendarBooking> childCalendarBookings =
 			calendarBooking.getChildCalendarBookings();
 
+		WorkflowInstanceLinkLocalService workflowInstanceLinkLocalService =
+			_workflowInstanceLinkLocalServiceSnapshot.get();
+
 		jsonObject.put(
 			"hasChildCalendarBookings", childCalendarBookings.size() > 1
 		).put(
 			"hasWorkflowInstanceLink",
-			_workflowInstanceLinkLocalService.hasWorkflowInstanceLink(
+			workflowInstanceLinkLocalService.hasWorkflowInstanceLink(
```

Comment:
@W2jx Please use `JSONObject.put(key, supplier)` and get the service in the supplier whenever fits the usage for the variables you replaced to make the long chain clearer.